src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -1049,7 +1049,8 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
             }
           }
           debuglog(s"specialized overload $om for ${overriding.name.decode} in ${pp(env)}: ${om.info}")
-          if (overriding.isAbstractOverride) om.setFlag(ABSOVERRIDE)
+          om.setFlag(overriding.flags & (ABSOVERRIDE | SYNCHRONIZED))
+          om.withAnnotations(overriding.annotations.filter(_.symbol == ScalaStrictFPAttr))
```

Comment:
OK, I see now.

```
class Foo extends (() => Unit) {
  def apply(): Unit = synchronized(println())
}
```

the symbol for `apply$mcV$sp` in `Foo` is cloned from the symbol of `apply` in `Function0`, not in `Foo`. I.e., a few lines above, we have `val om = specializedOverload(clazz, overridden, env)`. We cannot easily clone `overriding` instead, as this one no longer refers to type params..